joblib/func_inspect.py

```diff
@@ -156,6 +160,47 @@ def get_func_name(func, resolv_alias=True, win_characters=True):
     return module, name
 
 
+def getfullargspec(func):
+    """Compatibility function to provide inspect.getfullargspec in Python 2"""
```

Comment:
Maybe add a comment to state that this will be rewritten with a backport of Python 3 `Signature` once we drop support for Python 2.6 that would impose an additional backport of `OrderedDict`.